all-sdk/external-client-configuration.md

```diff
@@ -0,0 +1,551 @@
+
+    temporal workflow list
+
+Similarly, if you're using an SDK, you can load the config and just switch namespaces in code. So you can have:
+
+```python
+config = envconfig.load_client_connect_config()
+client = await Client.connect(namespace="my-specific-namespace", **config)
+```
+
+## Specification
+
+### Values and File Format
+
+* The format is a TOML object with the following fields:
+  * `profile` - Object with keys as profile names. The default profile should be named `default`.
+    * `<name>` - Profile object with configuration values. Profile names are case-insensitive. 💭 Why? They are needed
+      for environment variables which are case insensitive. It is a validation error to have a config file with two
+      separately-cased profile names that are equal case-insensitively.
+      * `address` - Client address, aka gRPC "host:port". This cannot be a URL, it must be `host:port`.
+      * `namespace` - Client namespace.
+      * `api_key` - Client API key.
+      * `tls` - A boolean (true is same as empty object) _or_ an object. Note the default for this value is dependent
+        upon other settings here. 💭 Why? We regret not making TLS the default in the past, so we will make it default
+        if `api_key` is present. If TLS is an object, it can have the following possible fields:
+        * `client_cert_path` - File path to mTLS client cert. Mutually exclusive with `client_cert_data`.
+        * `client_cert_data` - Cert data. Mutually exclusive with `client_cert_path`.
+        * `client_key_path` - File path to mTLS client key. Mutually exclusive with `client_key_data`.
+        * `client_key_data` - Key data. Mutually exclusive with `client_key_path`.
+        * `server_ca_cert_path` - File path to server CA cert. Mutually exclusive with `server_ca_cert_data`.
+        * `server_ca_cert_data` - CA cert data. Mutually exclusive with `server_ca_cert_path`.
+        * `server_name` - Override SNI name when connecting to server.
+        * `disable_host_verification` - Boolean to disable verifying TLS server host. May not be available in all
+          Temporal clients. 💭 Why? Not all SDKs offer this feature today.
+      * `codec` - Remote codec information to use for all encoding/decoding of payloads. May not be available in all
+        Temporal clients. 💭 Why? Not all SDKs allow use of a remote codec today.
+        * `endpoint` - Endpoint to the remote codec.
+        * `auth` - Authorization header for the remote codec.
+      * `grpc_meta` - Object representing HTTP headers. Values must be strings for now (can discuss arrays later).
+        ❓ Should we call this `headers`? We do in some cases and not others.
+  * 💭 Why TOML?
+    * Original document was JSON, but team discussion decided that the benefit to humans of TOML outweighed the concerns
+      of making SDKs have extensions with TOML support.
+* For settings that accept files above, they are relative to the current working directory.
```

Comment:
Need to clarify that the config file must hold absolute paths and the CLI will always write absolute paths.

Reply:
Ah, I clarified this elsewhere but not here. Will do.